### ISSUE TITLE
const assertion fails on ternary expression containing literals

### ISSUE THREAD

**Issue by NimaiMalle** [microsoft/TypeScript#49957]
# Bug Report

TypeScript's `as const` assertion yields the correct result _and_ throws an error as well, when used on a ternary operation, even though it's all literals.  The inconsistency between the type seeming to be as expected, while also having an error, made be think to file this as a bug.

### 🔎 Search Terms

const, const assertion, ternary, literal

### 🕗 Version & Regression Information

Just ran into this, but can reproduce back to 3.5.x, and Nightly.

This is the behavior in every version I tried, and I reviewed the FAQ for entries about const and as-const.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220719#code/GYVwdgxgLglg9mABMOcAUAPAXGEBbAIwFMAnASiwGcoSYwBzRAbwChFEIFrEA3AQwA2IIogC8iTIgB8iAIwAGeYgD8iAOQC+JekTVZ1lPIIFqyiPpQ5coAbjaIA9A-bsAesvskiUECST8hIjsAXxYgA)

### 💻 Code

```ts
function foo(x:number):string {
  const value = (x > 100 ? 'large': 'small') as const;
  //    ^?
  return value;
}
```

### 🙁 Actual behavior

The type was correctly discerned, but an error was also reported.

### 🙂 Expected behavior

The type would be the union of possible string literals without error.

**Comment by fatcerberus** [comment]
Worth noting that the `as const` doesn't actually affect the type of the expression, it's `"large" | "small"` even without it.

[Playground](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220719#code/GYVwdgxgLglg9mABMOcAUAPAXGEBbAIwFMAnASiwGcoSYwBzRAbwChFEIFrEA3AQwA2IIogC8iDIgB8iAIwAGeYgD8iAOQC+JekTVZ1lPIIFqA3G0QB6S+3YA9ZRZJEoIEkn5Ci5gL4sgA)

**Comment by NimaiMalle** [comment]
By the **"Working as intended"** label, am I to understand that `as const` is not supposed to work with literals, if they're involved in a ternary operation?  Or maybe it's redundant, and therefore an error?  Hoping to understand.

**Comment by fatcerberus** [comment]
`as const` only works on bare literals, not arbitrary expressions--even if they always evaluate to literals.  It tells the compiler not to widen a literal to its base type (or, for an object/array literal, not to widen its properties).

```
'const' assertions can only be applied to references to enum members, or string, number, boolean, array, or object literals.
```

A ternary expression containing string literals is not itself a string literal.
